networkmanager/src/main/java/com/dezen/riccardo/networkmanager/database_dictionary/PeerDao.java

```diff
@@ -0,0 +1,27 @@
+package com.dezen.riccardo.networkmanager.database_dictionary;
+
+import androidx.room.Dao;
+import androidx.room.Delete;
+import androidx.room.Insert;
+import androidx.room.OnConflictStrategy;
+import androidx.room.Query;
+import androidx.room.Update;
+
+/**
+ * @author Giorgia Bortoletti
+ */
+@Dao
+public interface PeerDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    public void add(PeerEntity... peerEntities);
+    @Update
+    public void update(PeerEntity... peerEntities);
+    @Delete
+    public void remove(PeerEntity... peerEntities);
+
+    @Query("SELECT * FROM peerentity")
+    public PeerEntity[] getAll();
+
+    @Query("SELECT 1 FROM peerentity WHERE address=:address")
+    public boolean contains(String address);
```

Comment:
"1" isn't hardwiring in this case. The method has the purpose to return true/false if there is a entity with the address passed as parameter but in a query you can't write "true"/"false" so this value "1" stands for "true".
For the name "peerentity" or "address": the first is the name of database and I'll replace that with a constant; the second is written with two different scope. "WHERE **address**:=_address_ ": **address** is the name of Peer's primary key and _address_ is the parameter of the method. So I don't think it's the best solution replace them because it would be like saving in a variable the name of another variable.

Reply:
I see...
I proposed this, because it looks like @ Query wants String value only...
And I agree with you about having a constant with same name as the value.
In other cases and scenarios (tip for the future) you might consider the enum type ;)

Reply:
Excuse me I don't mean to barge in but I read something because I'm receiving e-mail notifications on all comments. Because of the Room persistence library, which we are using to simplify database management, database access methods have to be defined in an interface, in order to be instantiated by the library iteself, and their annotations don't seem to be able to receive constants or variables as part of their value, not even from an enclosing class, as shown in the picture
![Cattura](https://user-images.githubusercontent.com/56487105/69451905-bf8e0a80-0d60-11ea-84b1-081006c10efb.PNG)

Reply:
Room library makes db management very simple and safe, but I'm yet to find an alternative to hardwiring the queries